pkg/controller/factory/controller_context.go

```diff
@@ -0,0 +1,114 @@
+package factory
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// ctx provide access to controller name, queue and event recorder.
+type controllerContext struct {
+	queue          workqueue.RateLimitingInterface
+	eventRecorder  events.Recorder
+	controllerName string
```

Comment:
as written before, don't like to have controllerName here. Just make it a string (first arg) to the controller factory constructor.

Reply:
removed name, we don't need it anyway